fog/sample-paykit/src/cached_tx_data/mod.rs

```diff
@@ -127,10 +136,28 @@ impl CachedTxData {
     /// then we would have to track if we did that. Then this `min`
     /// expression would also take the min of outstanding missed blocks.
     pub fn get_num_blocks(&self) -> BlockCount {
-        min(
-            self.rng_set.get_highest_processed_block_count(),
-            self.key_image_data_completeness,
-        )
+        if self.missed_block_ranges.is_empty() {
+            min(
+                self.rng_set.get_highest_processed_block_count(),
+                self.key_image_data_completeness,
+            )
+        } else {
+            *vec![
```

Comment:
```suggestion
            *[
```
I think this would also work if it's just an array on the stack and not a vec